src/sql/src/pure.rs

```diff
@@ -0,0 +1,184 @@
+// Copyright Materialize, Inc. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+//! Statement purification.
+
+use std::collections::HashMap;
+
+use failure::{bail, ResultExt};
+use tokio::io::AsyncBufReadExt;
+
+use repr::strconv;
+use sql_parser::ast::{AvroSchema, Connector, CsrSeed, Format, Ident, Statement};
+
+use crate::kafka_util;
+use crate::normalize;
+
+/// Removes dependencies on external state from `stmt`: inlining schemas in
```

Comment:
Maybe add a note that this is "where possible?" e.g. we do not remove the dependency on external state like certificate files.